graphbolt/src/index_select.cc

```diff
@@ -21,6 +22,25 @@ torch::Tensor IndexSelect(torch::Tensor input, torch::Tensor index) {
   return input.index({index.to(torch::kLong)});
 }
 
+torch::Tensor DiskIndexSelect(
+    std::string path, torch::Tensor index, torch::ScalarType dtype) {
+#ifdef __linux__
+  storage::OnDiskNpyArray arr(path);
+  return arr.index_select_iouring({index.to(torch::kLong)}, dtype);
+#else
+  return torch::empty({0});
+#endif
+}
+
+torch::Tensor DiskFeatureShape(std::string path) {
```

Comment:
If we keep an open handle, then we could cache the size of the tensor instead of rereading it everytime.